crates/cxx-qt-gen/src/generator/structuring/qobject.rs

```diff
@@ -31,4 +33,20 @@ impl<'a> StructuredQObject<'a> {
             signals: vec![],
         }
     }
+
+    pub fn method_lookup(&self, id: &Ident) -> Result<Name> {
+        let method = self
+            .methods
+            .iter()
+            .find(|method| method.name.rust_unqualified() == id);
+
+        method
+            .map(|method| method.name.clone())
+            .ok_or_else(|| Error::new_spanned(id, format!("Method with name '{id}' not found!")))
+    }
```

Comment:
this could all be simplified into one glorious chain of items with no `let` as well ? :tada: 

```rust
   pub fn method_lookup(&self, id: &Ident) -> Result<Name> {
        self
            .methods
            .iter()
            .find(|method| method.name.rust_unqualified() == id)
            .map(|method| method.name.clone())
            .ok_or_else(|| Error::new_spanned(id, format!("Method with name '{id}' not found!")))
    }
```

Reply:
or even:
```rust
   pub fn method_lookup(&self, id: &Ident) -> Result<Name> {
        self
            .methods
            .iter()
            .map(|method| &method.name)
            .find(|name| name.rust_unqualified() == id)
            .cloned()
            .ok_or_else(|| Error::new_spanned(id, format!("Method with name '{id}' not found!")))
    }
```

Reply:
Done